File: backend/app/schemas/billing.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class CheckoutRequest(BaseModel):
    plan: Literal["monthly", "annual"]


class CheckoutResponse(BaseModel):
    checkout_url: str


class PortalResponse(BaseModel):
    portal_url: str


class BillingStatusResponse(BaseModel):
    plan_type: str | None
    status: str | None
    current_period_end: datetime | None
    cancel_at_period_end: bool
    has_access: bool

File: frontend/types/billing.ts
export type BillingPlan = 'monthly' | 'annual';

export type CheckoutRequest = {
  plan: BillingPlan;
};

export type CheckoutResponse = {
  checkout_url: string;
};

export type BillingStatusResponse = {
  has_active_access: boolean;
  subscription_status: string | null;
  plan_type: string | null;
  current_period_end: string | null;
  cancel_at_period_end: boolean;
};

export type CustomerPortalResponse = {
  portal_url: string;
};
